src/kde_analysis/eval_m1m2chieff_poiss_weights.py

```diff
@@ -179,52 +186,96 @@ def get_rate_m_chieff2D(m1_query, m2_query, Rate):
     if iter_name not in hdf:
         print(f"Iteration {it} not found in file.")
         continue
-
     group = hdf[iter_name]
     if 'bootstrap_weights' in group:
         weighted = True
-        weights = group['bootstrap_weights'][:]
+        poisson_weights = group['bootstrap_weights'][:]
+        assert min(poisson_weights) > 0, "Some bootstrap weights are non-positive!"
+    
+    # Check if VT weighting should be used
+    if 'rwvt_vals' in group:
+        vt_weights = True
+        vt_vals = group['rwvt_vals'][:]
+    
     samples = group['rwsamples'][:]
     alpha = group['alpha'][()]
     bwx = group['bwx'][()]
     bwy = group['bwy'][()]
     bwz = group['bwz'][()]
-
-    # Remove samples with zero bootstrap weight as they may have bad behaviour in
-    # adaptive KDE (due to extremely small pilot density at the sample location)
-    # and have no effect on the KDE, and to reduce compute cost
-    samples = samples[weights > 0., :]
-    weights = weights[weights > 0.]
-
+    
     # Create the KDE with mass symmetry
     m1 = samples[:, 0]  # First column corresponds to m1
     m2 = samples[:, 1]  # Second column corresponds to m2
     cf = samples[:, 2]
     samples2 = np.vstack((m2, m1, cf)).T
     # Combine both samples into one array
     symmetric_samples = np.vstack((samples, samples2))
-    # Double the weights for bootstrap uncertainty
-    weights = np.concatenate((weights, weights)) if weighted else None
-
-    train_kde = ad.AdaptiveBwKDE(
-        symmetric_samples,
-        weights,
-        input_transf=('log', 'log', 'none'),
-        stdize=True,
-        rescale=[1/bwx, 1/bwy, 1/bwz],
-        alpha=alpha
-    )
-
-    # Evaluate KDE on the evaluation samples
+    
+    # Determine weights based on vt_weights flag
+    if weighted:
+        if vt_weights:
+            weights_over_VT = poisson_weights / vt_vals
+            weights = np.tile(weights_over_VT, 2)
+        else:
+            weights = np.tile(poisson_weights, 2)
+    else:
+        weights = None
+    
+    # Check if per-point bandwidth exists
+    use_variable_bw = 'perpoint_bws' in group
```

Comment:
This variable is only used once, so there is no need for it. Just use `if 'perpoint_bws' in group` in the next statement